docs/installation/deployment/kubernetes.rst

```diff
@@ -0,0 +1,280 @@
+
+    `Kubernetes`_ is an open-source system for automating deployment,
+    scaling, and management of containerized applications.
+
+    -- kubernetes.io
+
+
+Kubernetes (K8s) is one of the main supported targets for deployment. We
+provide a fully automated deployment toolset suitable for DevOps engineers
+and/or system administrators.
+
+This documentation will sketch out the architecture, document the initial
+requirements and teach you how to deploy on K8s.
+
+This setup is tested against a Google Cloud Kubernetes cluster with 4vCPU and
+15G of memory.
+
+Application architecture
+========================
+
+Typically, your K8s cluster provider will set up a load balancer in front
+of your cluster, directing all the traffic to services in the cluster.
+
+DNS is set up so that your domain name(s) point to the IP address of the load
+balancer.
+
+The load balancer will then pass the traffic to an *Ingress* in the cluster,
+which will route the traffic to the backend services.
+
+All applications make use of caching via Redis cache databases.
+
+Open Zaak
+---------
+
+Traffic from the ingress will enter Open Zaak at one or multiple `NGINX`_
+reverse proxies. NGINX will performs one of two possible tasks:
+
+* pass the request to the Open Zaak application or
+* send the binary data of uploaded files to the client
+
+NGINX exists to serve uploaded files in a secure and performant way, reducing
+load on the Open Zaak application.
+
+All other requests are passed to the Open Zaak application.
+
+Open Notificaties
+-----------------
+
+Traffic from the ingress is directly passed to the Open Notificaties
+application. Open Notificaties then communicates with async workers (using
+RabbitMQ) to distribute the notifications to all the relevant subscribers.
+
+Environment requirements
+========================
+
+Before you begin, you will need:
+
+* a kubernetes cluster that you can access, this means you need a valid
+  ``~/.kube/config`` file. You can override which kube config to use by setting
+  the ``KUBECONFIG`` environment variable if you manage multiple clusters.
+
+  If you are dealing with jump/bastion hosts, complicated firewalls... Please
+  contact your provider on how you can access your cluster from your local
+  machine.
+
+* a PostgreSQL (10 or 11) database server with credentials:
+
+    * a database hostname that you can reach from your local machine
+    * a database hostname that can be reached from your K8s cluster (possibly
+      the same as above)
+    * the username of a superuser (typically ``postgres``)
+    * a password for the superuser
+    * credentials for the Open Zaak database and Open Notificaties database
+
+* A persistent-volume storage class supporting ``ReadWriteMany``. Contact your
+  provider to see if they offer it. If this is not an option, you can use a
+  ``ReadWriteOnce`` storage class and set up an NFS-server around it, but this
+  will likely have slower performance.
+
+  On Google Cloud, you can use:
```

Comment:
Perhaps full gcloud command list to setup the needed requirements for production.

Reply:
Added those to the introduction/requirements section